src/main/kotlin/app/revanced/patches/youtube/layout/miniplayer/MiniplayerPatch.kt

```diff
@@ -0,0 +1,356 @@
+                "19.16.39" // Earliest supported version with modern miniplayers.
+            ]
+        )
+    ]
+)
+@Suppress("unused")
+object MiniplayerPatch : BytecodePatch(
+    setOf(GetFormFactorFingerprint,
+        MiniplayerDimensionsCalculatorParentFingerprint,
+        MiniplayerResponseModelSizeCheckFingerprint,
+        MiniplayerOverrideFingerprint,
+        MiniplayerModernConstructorFingerprint,
+        MiniplayerModernViewParentFingerprint,
+        YouTubePlayerOverlaysLayoutFingerprint
+    )
+) {
+    private const val INTEGRATIONS_CLASS_DESCRIPTOR = "Lapp/revanced/integrations/youtube/patches/MiniplayerPatch;"
+
+    override fun execute(context: BytecodeContext) {
+        AddResourcesPatch(this::class)
+
+        // Modern mini player is only present and functional in 19.15+
+        // Of note, some modern miniplayer code is present in 19.14 but the feature is not complete.
+        val isPatchingLegacy = ytOutlinePictureInPictureWhite24 < 0
+
+        SettingsPatch.PreferenceScreen.PLAYER.addPreferences(
+            PreferenceScreen(
+                key = "revanced_miniplayer_screen",
+                sorting = Sorting.UNSORTED,
+                preferences =
+                if (isPatchingLegacy) {
+                    setOf(
+                        ListPreference(
+                            "revanced_miniplayer_type",
+                            summaryKey = null,
+                            entriesKey = "revanced_miniplayer_type_legacy_entries",
+                            entryValuesKey = "revanced_miniplayer_type_legacy_entry_values"
+                        )
+                    )
+                } else {
+                    setOf(
+                        ListPreference(
+                            "revanced_miniplayer_type",
+                            summaryKey = null,
+                            entriesKey = "revanced_miniplayer_type_19_15_entries",
+                            entryValuesKey = "revanced_miniplayer_type_19_15_entry_values"
+                        ),
+                        SwitchPreference("revanced_miniplayer_hide_expand_close"),
+                        SwitchPreference("revanced_miniplayer_hide_subtext"),
+                        SwitchPreference("revanced_miniplayer_hide_rewind_forward"),
+                        TextPreference("revanced_miniplayer_opacity", inputType = InputType.NUMBER)
+                    )
+                }
+            )
+        )
+
+        // region Enable tablet miniplayer.
+
+        MiniplayerOverrideNoContextFingerprint.resolve(
+            context,
+            MiniplayerDimensionsCalculatorParentFingerprint.resultOrThrow().classDef
+        )
+        MiniplayerOverrideNoContextFingerprint.resultOrThrow().mutableMethod.apply {
+            findReturnIndices().forEach { index -> insertLegacyTabletOverride(index) }
+        }
+
+        // endregion
+
+
+        // region Legacy tablet hooks.
```

Comment:
In conjunction with my review above, this could be renamed "Old tablet" instead, otherwise, dismiss the review above. Also, is this referring to the Miniplayer? If so, this should say:


```suggestion
        // region Legacy tablet Miniplayer hooks.
```